### PR TITLE
Add precursor-mz match score

### PR DESCRIPTION
**Pull request by florian-huber**
- fixed minor bug in ``add_parent_mass()`` : now finds precursor_mz from either pepmass or precursor_mz field.
- added ``PrecursorMzMatch`` which is analog to ``ParentMassMatch`` (renamed here from ``ParentmassMatch``).
- include option to match based on relative ``ppm`` tolerance

### CODE REVIEW

File: matchms/similarity/PrecursormzMatch.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import numba
+import numpy
+from matchms.typing import SpectrumType
+from .BaseSimilarity import BaseSimilarity
+
+
+class PrecursormzMatch(BaseSimilarity):
+    """Return True if spectrums match in precursor m/z (within tolerance), and False otherwise.
+
+    Example to calculate scores between 2 spectrums and iterate over the scores
```
[the remark this answers is not in the document]
Comment (florian-huber) [reply matchms/similarity/PrecursormzMatch.py]: True, that's better.

File: matchms/similarity/PrecursormzMatch.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    is_commutative = True
+
+    def __init__(self, tolerance: float = 0.1, tolerance_type: str = "Dalton"):
+        """
+        Parameters
+        ----------
+        tolerance
+            Specify tolerance below which two m/z are counted as match.
+        tolerance_type
+            Chose between fixed tolerance in Dalten (="Dalton") or a relative difference
```
Comment (sverhoeven) [inline matchms/similarity/PrecursormzMatch.py]: Typo
```suggestion
            Chose between fixed tolerance in Dalton (="Dalton") or a relative difference
```

File: matchms/similarity/__init__.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 from .IntersectMz import IntersectMz
 from .ModifiedCosine import ModifiedCosine
 from .ParentmassMatch import ParentmassMatch
+from .PrecursormzMatch import PrecursormzMatch
```
Comment (sverhoeven) [inline matchms/similarity/__init__.py]: The doc string above lists a couple of matcher classes. Maybe nice to add `PrecursormzMatch` next to `ParentmassMatch` class.

File: matchms/similarity/PrecursormzMatch.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                assert precursormz is not None, "Missing precursor m/z."
+                precursors.append(precursormz)
+            return numpy.asarray(precursors)
+
+        precursors_ref = collect_precursormz(references)
+        precursors_query = collect_precursormz(queries)
+        if is_symmetric and self.type == "Dalton":
+            return precursormz_scores_symmetric(precursors_ref, precursors_query, self.tolerance).astype(bool)
+        if is_symmetric and self.type == "ppm":
+            return precursormz_scores_symmetric_ppm(precursors_ref, precursors_query, self.tolerance).astype(bool)
```
Comment (sverhoeven) [inline matchms/similarity/PrecursormzMatch.py]: This line does not seem to be tested, see https://sonarcloud.io/code?id=matchms_matchms&pullRequest=156&selected=matchms_matchms%3Amatchms%2Fsimilarity%2FPrecursormzMatch.py&line=119

Can a test be added for it?

File: matchms/filtering/add_parent_mass.py [hunk 5]
```diff
@@ -17,7 +18,9 @@ def add_parent_mass(spectrum_in: SpectrumType) -> SpectrumType:
         try:
             charge = spectrum.get("charge")
             protons_mass = PROTON_MASS * charge
-            precursor_mz = spectrum.get("pepmass")[0]
+            precursor_mz = spectrum.get("precursor_mz", None)
```
Comment (sverhoeven) [inline matchms/filtering/add_parent_mass.py:21]: Nice, it now behaves as the reverse of the `add_precursor_mz()` filter.

**Review by sverhoeven — CHANGES_REQUESTED** [review]
The `IntersectMz` class uses capitailized `Mz`, but `ParentmassMatch` and `PrecursormzMatch` classes uses lower case `mass`/`mz` resp.. This seems inconsistent to me. Either rename to `Intersectmz` or rename to `ParentMassMatch` + `PrecursorMzMatch`.

File: matchms/similarity/__init__.py (hunk 3 above)
Comment (florian-huber) [reply matchms/similarity/__init__.py]: Good point. Added it to the docstring.

File: matchms/similarity/PrecursormzMatch.py (hunk 4 above)
Comment (florian-huber) [reply matchms/similarity/PrecursormzMatch.py]: I now added a test for this as well.

File: matchms/filtering/add_parent_mass.py (hunk 5 above)
Comment (florian-huber) [reply matchms/filtering/add_parent_mass.py:21]: Thanks!

**Comment by florian-huber** [comment]
> 
> 
> The `IntersectMz` class uses capitailized `Mz`, but `ParentmassMatch` and `PrecursormzMatch` classes uses lower case `mass`/`mz` resp.. This seems inconsistent to me. Either rename to `Intersectmz` or rename to `ParentMassMatch` + `PrecursorMzMatch`.

Switched to ``ParentMassMatch`` and `PrecursorMzMatch`.

File: matchms/similarity/__init__.py (hunk 3 above)
Comment (sverhoeven) [reply matchms/similarity/__init__.py]: thanks

File: matchms/similarity/PrecursormzMatch.py (hunk 4 above)
Comment (sverhoeven) [reply matchms/similarity/PrecursormzMatch.py]: great 100% coverage

**Review by sverhoeven — APPROVED** [review]

**Comment by florian-huber** [comment]
Thanks a lot for the review Stefan!

**Pull request merged**